third_party/terraform/tests/data_source_google_folder_organization_policy_test.go

```diff
@@ -0,0 +1,93 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccDataSourceGoogleFolderOrganizationPolicy_basic(t *testing.T) {
+	folder := acctest.RandomWithPrefix("tf-test")
+	org := getTestOrgFromEnv(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleFolderOrganizationPolicy_basic(org, folder),
+				Check: testAccDataSourceGoogleOrganizationPolicyCheck(
+					"data.google_folder_organization_policy.data",
+					"google_folder_organization_policy.resource"),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGoogleOrganizationPolicyCheck(dataSourceName string, resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		ds, ok := s.RootModule().Resources[dataSourceName]
+		if !ok {
+			return fmt.Errorf("root module has no resource called %s", dataSourceName)
+		}
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", resourceName)
+		}
+
+		dsAttr := ds.Primary.Attributes
+		rsAttr := rs.Primary.Attributes
+
+		cloudFuncAttrToCheck := []string{
+			"name",
+			"folder",
+			"constraint",
+			"version",
+			"list_policy",
+			"restore_policy",
+			"boolean_policy",
+		}
+
+		for _, attr := range cloudFuncAttrToCheck {
+			log.Printf("ZOMG checking [%s] -> [%s]\n", dsAttr[attr], rsAttr[attr])
```

Comment:
> ZOMG

lol

Reply:
🤦‍♂️ really need to write a pre-submit for myself to catch my logging statements.